tools/perform_upgrade.sh

```diff
@@ -23,7 +23,7 @@ set -e
 BOARD="$1"
 
 ./deploy.py --board=$BOARD --opensk --programmer=none
-python3 -m tools.deploy_partition --board=$BOARD
+python3 -m tools.deploy_partition --board=$BOARD "$2"
```

Comment:
nit: This is not very clean, but if python accepts empty arguments then why not. I see at least 2 solutions:

```
BOARD="$1"
shift

./deploy.py ...
python3 -m ... --board="${BOARD}" "$@"
```
(note that Google style for bash is `"${var}"`, same for deploy.py line)

```
python3 ... --board="${BOARD}" $2
```
(note the absence of double quotes, this means that if $2 is empty, no extra argument is passed, while double quotes would pass an empty argument)

Reply:
Seems like I didn't actually test the empty argument case after introducing `"$2". It would indeed not have worked. I agree it doesn't look clean, but seemed pretty flexible to me, since this script doesn't really do more than forward everything to Python.

Also I made it so that you also deploy with Vendor HID if you upgrade over Vendor HID, because that should be the most likely use case.

Reply:
`$2` is already much cleaner than `"$2"`, the only difference with `"$@"` is that it only supports one optional flag instead of an arbitrary number.

Reply:
Since we don't actually check the content, this could actually be a feature?

Reply:
It's definitely a possible choice. It just has more than 90% chance of being useless compared to the other one. But since it doesn't matter for now we can always fix it in the future (at 90% chance).